pipeline/runspec_test.go

```
@@ -0,0 +1,217 @@
+	}
+	// The source still landed: one artifact failing must not cost the others.
+	if manifest.SourceFile == "" {
+		t.Error("source was not written despite being independent of the failure")
+	}
+	if len(manifest.Inputs) != 0 {
+		t.Error("manifest lists an input that was never stored")
+	}
+}
+
+func TestDigestIsStable(t *testing.T) {
+	if digest([]byte("abc")) != digest([]byte("abc")) {
```

golangci-lint (2.12.2)

[error] SA4000: identical expressions on the left and right side of the '!=' operator

(staticcheck)